lib/Sema/CSSimplify.cpp

```diff
@@ -7690,8 +7690,16 @@ ConstraintSystem::simplifyKeyPathConstraint(
       return SolutionKind::Solved;
 
     // If the root type has been bound to a hole, we cannot infer it.
-    if (getFixedTypeRecursive(rootTy, /*wantRValue*/ true)->isHole())
+    if (getFixedTypeRecursive(rootTy, /*wantRValue*/ true)->isHole()) {
+      auto *fix = SpecifyKeyPathRootType::create(
+          *this, getConstraintLocator(locator));
+      recordFix(fix);
+      
+      if (auto *rootTyVar = rootTy->getAs<TypeVariableType>()) {
+        assignFixedType(rootTyVar, getASTContext().TheUnresolvedType);
```

Comment:
I don't understand what this is trying to do. If `isHole()` returns `true` it means that `rootTy` is already bound to `UnresolvedType`....